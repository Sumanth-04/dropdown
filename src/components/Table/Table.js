import React from 'react';
import PropTypes from 'prop-types';
import RcTable from 'rc-table';

class Table extends React.Component {

	render() {

		const {
			columns,
			data
		} = this.props;

		return (
			<RcTable
				columns={columns}
				data={data}
				{...this.props}
			/>
		)
	}
}

Table.propTypes = {
	columns: PropTypes.array,
	data: PropTypes.array,
};

export default Table;