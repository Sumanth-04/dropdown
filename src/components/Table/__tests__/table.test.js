import React from "react";
import renderer from "react-test-renderer";
import Table from "../Table";

it("Table Renders with data and column props", () => {
  const tree = renderer
    .create(
      <Table
         columns={[{
          title: 'Name',
          dataIndex: 'name',
          key: 'name',
          width: 100,
        }]} 
        data={[
          { name: 'Jack'}
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
