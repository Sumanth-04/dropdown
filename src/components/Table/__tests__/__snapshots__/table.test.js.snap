// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Renders with data and column props 1`] = `
<div
  className="rc-table"
>
  <div
    className="rc-table-container"
  >
    <div
      className="rc-table-content"
      onScroll={[Function]}
      style={Object {}}
    >
      <table
        style={
          Object {
            "tableLayout": "auto",
          }
        }
      >
        <colgroup>
          <col
            style={
              Object {
                "minWidth": 100,
                "width": 100,
              }
            }
          />
        </colgroup>
        <thead
          className="rc-table-thead"
        >
          <tr>
            <th
              className="rc-table-cell"
              colSpan={null}
              rowSpan={null}
              style={Object {}}
            >
              Name
            </th>
          </tr>
        </thead>
        <tbody
          className="rc-table-tbody"
        >
          <tr
            className="rc-table-row rc-table-row-level-0"
            onClick={[Function]}
            style={Object {}}
          >
            <td
              className="rc-table-cell"
              colSpan={null}
              rowSpan={null}
              style={Object {}}
            >
              Jack
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
